File: TPs/TP1/TP1-Q1/palindromo.java
/**
 * palindromo
 */

public class palindromo {

    public static boolean fim(String str){

        boolean resultado = false;

        if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){

            resultado = true;

        }

        return resultado;

    }
    
    public static boolean ehPalindromo(String str){

        boolean resultado = true;
        int length = str.length();
        int i = 0;
        int j = 0;

        for(i = 0,j = length - 1; i < length/2; i++,j--){

            if(str.charAt(i) != str.charAt(j)){
                resultado = false;
                i = length;
            }

        }

        return resultado;

    }
    
    public static void main(String[] args) {
        
        String str;

        do{

            str = MyIO.readString();

            if(!fim(str)){

                if(ehPalindromo(str)){
                    System.out.println("SIM");
                }else{
                    System.out.println("NAO");
                }

            }

        }while(!fim(str));
        
    }

}

File: TPs/TP1/TP1-Q2/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool palindromo(char* string)
{
    int length = strlen(string);
    bool resultado = true;
    int i = 0, j = 0;

    for(i = 0,j = length-1; i < length/2;i++, j--)
    {

        if(string[i] != string[j])
        {
            resultado = false;
            i = length;
        }
        
    }

    return resultado;
}

bool fim(char* string)
{
    bool fim = false;

    if(strcmp(string,"FIM") == 0)
    {
        fim = true;
    }

    return fim;
}

void limparBuffer()
{
    int n;
    while((n = getchar()) != '\n' && n != EOF);
}

int main()
{
    char str[1000];

    do
    {
        scanf("%[^\n\r]", str);
        limparBuffer();
        
        if(!fim(str))
        {
            if(palindromo(str))
            {
                printf("SIM\n");
            }else{
                printf("NAO\n");
            }
        }

    }while(!fim(str));

    return 0;
}